bin/base64

```diff
@@ -38,6 +40,7 @@ if (defined $opt{'o'}) {
     $out = *STDOUT;
 }
 if (defined $ARGV[0] && $ARGV[0] ne '-') {
+    die "$0: '$ARGV[0]' is a directory\n" if (-d $ARGV[0]);
```

Comment:
Instead of die-ing, warn and exit with an appropriate exit value.

Reply:
Also, instead of using $0, use the basename of $0. I have some examples in PROGRAMMING_STYLE.md